synthesizer/benches/utilities/object_store.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkvm_utilities::{FromBytes, TestRng, ToBytes};
+
+use crate::utilities::{construct_next_block, initialize_vm, split, BenchmarkBatch, BenchmarkTransactions, Operation};
+use anyhow::Result;
+use console::{
+    account::PrivateKey,
+    network::Testnet3,
+    program::{Entry, Identifier, Literal, Plaintext, Record},
+};
+use itertools::Itertools;
+use rand::Rng;
+use snarkvm_synthesizer::{ConsensusStorage, Transaction, VM};
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+pub struct ObjectStore {
+    root: PathBuf,
+}
+
+impl ObjectStore {
+    pub fn new(root: PathBuf) -> Result<Self> {
+        // Create the root directory if it does not exist.
+        if !root.try_exists()? {
+            std::fs::create_dir_all(&root)?;
+        }
+        Ok(Self { root })
+    }
+
+    pub fn get<O: FromBytes, P: AsRef<Path>>(&mut self, path: P) -> Result<O> {
+        let full_path = self.root.join(path);
+        let bytes = std::fs::read(full_path)?;
+        O::from_bytes_le(&bytes)
+    }
+
+    pub fn insert<O: ToBytes + FromBytes, P: AsRef<Path>>(&mut self, path: P, object: &O) -> Result<()> {
+        let full_path = self.root.join(path);
+        if !full_path.parent().expect("parent directory exists").exists() {
+            std::fs::create_dir_all(full_path.parent().expect("parent directory exists"))?;
+        }
+        let bytes = object.to_bytes_le()?;
+        std::fs::write(&full_path, bytes)?;
+        Ok(())
+    }
+
+    pub fn contains<P: AsRef<Path>>(&self, path: P) -> bool {
+        self.root.join(path).exists()
```

Comment:
I wonder if an interrupted run couldn't result in a file existing, but having mangled contents; that would cause `contains` to return `true`, but a call to `get` would fail; even though it's slower, it might be safer for this method to call `get` instead